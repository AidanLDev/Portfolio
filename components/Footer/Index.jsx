import React from 'react'
import Link from 'next/link'
import { FaArrowAltCircleUp } from 'react-icons/fa'
import styles from './styles.module.scss'

export default function Footer() {
  return (
    <footer className={styles.footerContainer}>
      <Link href="#nav-bar">
        <span>Back to top</span>
        <FaArrowAltCircleUp />
      </Link>
    </footer>
  )
}
