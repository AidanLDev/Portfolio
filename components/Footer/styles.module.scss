.footerContainer {
  padding: 2rem;
  display: flex;
  flex-direction: row-reverse;
  a {
    display: flex;
    gap: 1rem;
    svg {
      align-self: center;
    }
    span {
      font-size: 1.15em;
    }
    span:hover {
      text-decoration: underline;
    }
  }
}