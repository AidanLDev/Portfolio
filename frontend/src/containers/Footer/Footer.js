import React from "react";
import { motion } from "framer-motion";
import Button from "components/Button/Button";
import Icon from "@mdi/react";
import { mdiChevronUp } from "@mdi/js";
import LinkedinLogo from "assets/LinkedinLogo.png";
import sslImg from "assets/positivessl_trust_seal_md_167x42.png";

import styles from "./Style";

const Footer = props => {
  const backToTop = () => window.scrollTo(0, 0);

  const hoverScale = 1.2;
  const tapScale = 0.8;

  return (
    <div className={styles.footerWrapper}>
      <div className={styles.tooltip}>
        <span className={styles.tooltiptext}>Secure SSL Certificate</span>
        <img
          src={sslImg}
          alt="SSL Certificate"
          width="167"
          height="42"
        />
      </div>
      <div className={styles.tooltip}>
        <span className={styles.tooltiptext}>Linkedin Profile</span>
        <a
          href="https://www.linkedin.com/in/aidanlowson1/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <motion.img
            src={LinkedinLogo}
            alt="LinkedIn Logo"
            width="50"
            height="50"
            whileHover={{ scale: hoverScale }}
            whileTap={{ scale: tapScale }} />
        </a>
      </div>
      <motion.div onClick={backToTop} className={styles.backToTopWrapper} whileHover={{ scale: hoverScale }} whileTap={{ scale: tapScale }}>
        <Button>Back to top</Button>
        <Icon path={mdiChevronUp} size={1} color={"white"} />
      </motion.div>
    </div>
  );
};

export default Footer;
