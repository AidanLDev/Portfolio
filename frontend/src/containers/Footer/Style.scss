@import "pallet";
@import "variables";
@import "mixins";

.footerWrapper {
  width: 100%;
  height: 100px;
  background-color: #131211;
  display: flex;
  justify-content: space-around;
  align-items: center;

  .tooltip {
    @include tooltip;
  }

  .tooltip .tooltiptext {
    @include tooltipAndText;
    @include positionBottom;
  }

  .tooltip:hover .tooltiptext {
    @include tooptipHover;
  }

  .tooltip .tooltiptext::after {
    @include tooltipArrowBottom;
  }

  img {
    margin: 10px 0 0 0;
  }

  .backToTopWrapper {
    cursor: pointer;
    svg {
      padding: 0;
      position: inherit;
      display: initial;
    }
  }
}
