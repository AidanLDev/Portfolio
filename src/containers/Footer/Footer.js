import React from 'react';
import { motion } from 'framer-motion';
import Button from 'components/Button/Button';
import EmailIcon from '@material-ui/icons/Email';
import Icon from '@mdi/react';
import { mdiChevronUp } from '@mdi/js';
import LinkedinLogo from 'assets/LinkedinLogo.png';
import GithubLogo from 'assets/Githublogo.png';

import styles from './Style.module.scss';

const Footer = (props) => {
  const backToTop = () => window.scrollTo(0, 0);

  const hoverScale = 1.2;
  const tapScale = 0.8;

  return (
    <div className={styles.footerWrapper}>
      <div className={styles.tooltip}>
        <span className={styles.tooltiptext}>Email Aidan</span>
        <a href='mailto:dev@aidanlowson.com'>
          <motion.div
            whileHover={{ scale: hoverScale }}
            whileTap={{ scale: tapScale }}
          >
            <EmailIcon fontSize='large' />
          </motion.div>
        </a>
      </div>
      <div className={styles.tooltip}>
        <span className={styles.tooltiptext}>Aidans Github profile</span>
        <a
          href='https://github.com/AidanLDev'
          rel='noopener noreferrer'
          target='_blank'
        >
          <motion.img
            src={GithubLogo}
            alt='Github Logo'
            width='50'
            height='50'
            whileHover={{ scale: hoverScale }}
            whileTap={{ scale: tapScale }}
            className={styles.socialMedia}
          />
        </a>
      </div>
      <div className={styles.tooltip}>
        <span className={styles.tooltiptext}>Linkedin Profile</span>
        <a
          href='https://www.linkedin.com/in/aidanlowson1/'
          rel='noopener noreferrer'
          target='_blank'
        >
          <motion.img
            src={LinkedinLogo}
            alt='LinkedIn Logo'
            width='50'
            height='50'
            whileHover={{ scale: hoverScale }}
            whileTap={{ scale: tapScale }}
            className={styles.socialMedia}
          />
        </a>
      </div>
      <motion.div
        onClick={backToTop}
        className={styles.backToTopWrapper}
        whileHover={{ scale: hoverScale }}
        whileTap={{ scale: tapScale }}
      >
        <Button footer>Back to top</Button>
        <Icon
          path={mdiChevronUp}
          size={1}
          color={'white'}
          className={styles.mobileOnly}
        />
      </motion.div>
    </div>
  );
};

export default Footer;
