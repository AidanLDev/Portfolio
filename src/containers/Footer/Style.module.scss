@import 'styles/pallet.scss';
@import 'styles/variables.scss';
@import 'styles/mixins.scss';

.footerWrapper {
  width: 100%;
  height: 100px;
  background-color: #131211;
  display: flex;
  justify-content: space-around;
  align-items: center;

  .tooltip {
    @include tooltip;
  }

  .tooltip .tooltiptext {
    @include tooltipAndText;
    @include positionBottom;
  }

  .tooltip:hover .tooltiptext {
    @include tooptipHover;
  }

  .tooltip .tooltiptext::after {
    @include tooltipArrowBottom;
  }

  .socialMedia {
    @media (max-width: 450px) {
      width: 50px;
    }
  }

  img {
    margin: 10px 0 0 0;
  }

  .backToTopWrapper {
    span {
      color: white;
      padding: 0 0 0 10px;
    }
    cursor: pointer;
    svg {
      padding: 0;
      margin: -2px 0px 0px -15px;
      display: initial;
      position: absolute;
      @media (max-width: 480px) {
        display: none;
      }
    }
  }
}
