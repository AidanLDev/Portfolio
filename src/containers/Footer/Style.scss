@import "pallet";
@import "variables";
@import "mixins";

.footerWrapper {
  width: 100%;
  background-color: black;
}