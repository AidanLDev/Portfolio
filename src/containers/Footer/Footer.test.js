import React from 'react';
import Footer from './Footer';
import { shallow } from 'enzyme';

it('Renders Footer Component Correctly', () => {
  const component = shallow(<Footer />);
  expect(component).toMatchSnapshot();
});
