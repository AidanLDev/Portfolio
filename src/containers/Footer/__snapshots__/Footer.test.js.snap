// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Footer Component Correctly 1`] = `
<div
  className="footerWrapper"
>
  <div
    className="tooltip"
  >
    <span
      className="tooltiptext"
    >
      Email Aidan
    </span>
    <a
      href="mailto:dev@aidanlowson.com"
    >
      <ForwardRef(MotionComponent)
        whileHover={
          Object {
            "scale": 1.2,
          }
        }
        whileTap={
          Object {
            "scale": 0.8,
          }
        }
      >
        <Memo(ForwardRef(EmailIcon))
          fontSize="large"
        />
      </ForwardRef(MotionComponent)>
    </a>
  </div>
  <div
    className="tooltip"
  >
    <span
      className="tooltiptext"
    >
      Aidans Github profile
    </span>
    <a
      href="https://github.com/AidanLDev"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(MotionComponent)
        alt="Github Logo"
        className="socialMedia"
        height="50"
        src="Githublogo.png"
        whileHover={
          Object {
            "scale": 1.2,
          }
        }
        whileTap={
          Object {
            "scale": 0.8,
          }
        }
        width="50"
      />
    </a>
  </div>
  <div
    className="tooltip"
  >
    <span
      className="tooltiptext"
    >
      Linkedin Profile
    </span>
    <a
      href="https://www.linkedin.com/in/aidanlowson1/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(MotionComponent)
        alt="LinkedIn Logo"
        className="socialMedia"
        height="50"
        src="LinkedinLogo.png"
        whileHover={
          Object {
            "scale": 1.2,
          }
        }
        whileTap={
          Object {
            "scale": 0.8,
          }
        }
        width="50"
      />
    </a>
  </div>
  <div
    className="backToTopWrapper"
    onClick={[Function]}
  >
    <Icon
      className="mobileOnly"
      color="white"
      horizontal={false}
      path="M7.41,15.41L12,10.83L16.59,15.41L18,14L12,8L6,14L7.41,15.41Z"
      rotate={0}
      size={1}
      spin={false}
      vertical={false}
    />
    <span>
      Back to top
    </span>
  </div>
</div>
`;
